source/PlayerInfo.cpp

```diff
@@ -2354,7 +2364,34 @@ void PlayerInfo::StepMissions(UI *ui)
 		else if(mission.CanComplete(*this))
 			RemoveMission(Mission::COMPLETE, mission, ui);
 		else if(mission.Destination() == GetPlanet() && !freshlyLoaded)
+		{
 			mission.Do(Mission::VISIT, *this, ui);
+			if(mission.IsUnique())
+				continue;
+			
+			// On visit dialogs are handled separately as to avoid a player
+			// getting spammed by on visit dialogs if they are stacking jobs
+			// from the same destination.
+			auto action = mission.GetAction(Mission::VISIT);
```

Comment:
Use `&` to avoid a copy. Probably we can avoid even needing to do this call at this block scope, since we only need the action's text if `visitText` is currently empty. We also don't need anything else about the action other than its text.